src/app/dao/tests/JoinerDAO.spec.ts

```diff
@@ -1,17 +1,462 @@
+            const fakeCreator: MinimalUser = { id: createdPart.creator.id, name: 'fake-jeanjaja' };
+            const result: Promise<void> = joinerDAO.set(createdPart.partId, { ...JoinerMocks.INITIAL, fakeCreator });
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should forbid non-verified user to create a joiner', async() => {
+            // Given an existing part.
+            // (Note that the non-verified user in practice can never have created the corresponding part,
+            // but this is an extra check just in case)
+            const partId: string = (await createPart({ signOut: true })).partId;
+
+            // and a non-verified user
+            const nonVerifiedUser: MinimalUser = await createUnverifiedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            // When the user tries to create the joiner
+            const result: Promise<void> = joinerDAO.set(partId, { ...JoinerMocks.INITIAL, nonVerifiedUser });
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should forbid to add themself to the candidates', async() => {
+            // Given a part and joiner, with the current user being the creator
+            const createdPart: CreatedPart = await createPart({ createJoiner: true });
+
+            // When trying to add itself to the candidates
+            const result: Promise<void> = joinerDAO.addCandidate(createdPart.partId, createdPart.creator);
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should allow to change other fields than candidates and partStatus to STARTED', async() => {
+            // Given a user that created a (non-started) part and joiner
+            const partId: string = (await createPart({ createJoiner: true })).partId;
+
+            const updates: Partial<Joiner>[] = [
+                { chosenOpponent: UserMocks.CANDIDATE_MINIMAL_USER },
+                { partStatus: PartStatus.CONFIG_PROPOSED.value },
+                { firstPlayer: FirstPlayer.CHOSEN_PLAYER.value },
+                { partType: PartType.BLITZ.value },
+                { maximalMoveDuration: 73 },
+                { totalPartDuration: 1001001 },
+            ];
+            for (const update of updates) {
+                // When modifying a field
+                const result: Promise<void> = joinerDAO.update(partId, update);
+
+                // Then it should succeed
+                await expectAsync(result).toBeResolvedTo();
+            }
+        });
+        describe('on started part', () => {
+            let partId: string;
+            beforeEach(async() => {
+                // Given a part that is started
+
+                // Creator creates the joiner
+                const createdPartId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+                partId = createdPartId;
```

Comment:
why the createdPartId var :p ?

Reply:
pour voir si tu es attentif :eyes: ! Corrigé